### ISSUE TITLE
Identifier is not optional in InsertDocument WOQL query

### ISSUE THREAD

**Issue by Neelterminusdb** [terminusdb/terminusdb#903]
**Describe the bug**
When i don't pass any identifier while inserting document using WOQL it gives the following error 

```
<body>

<h1>Internal server error</h1>

<p>
Unknown message: compilation_error('Failure to compile term insert_document(json{\'@type\':"Person",label:"Newperson"})')</p>

<address>
<a href="http://www.swi-prolog.org">SWI-Prolog</a> httpd at terminusdb-server-deployment-586f57bc7f-2jldk
</address>

</body>
</html>
Code: 500 url: https://cloud.terminusdb.com/cloudux/api/woql/cloudux/hellonew1/local/branch/main
```
The [woql.json](https://github.com/terminusdb/terminusdb/blob/main/src/terminus-schema/woql.json) file states that the identifier is optional.

**To Reproduce**
1. Insert document using WOQL query

**Expected behavior**
It should insert the document without passing any identifier

**Comment by spl** [comment]
This is true. I also discovered it last week. The problem, as I understand it, is that we allow it when parsing the JSON but don't have a clause to handle it when interpreting the WOQL in Prolog. I'm including the relevant clauses below. See the `insert_document` with 1 argument in `json_type_to_woql_ast` but no `insert_document` with 1 argument in `compile_wf`.

`src/core/query/json_woql.pl`:
```prolog
json_type_to_woql_ast('InsertDocument',JSON,WOQL,Path) :-
    _{document : Doc, 
      identifier : Doc_ID
     } :< JSON,
    !,   
    json_value_to_woql_ast(Doc, WDoc, [document
                                       |Path]),
    json_value_to_woql_ast(Doc_ID, WDoc_ID, [identifier
                                             |Path]),
    WOQL = insert_document(WDoc,WDoc_ID).
json_type_to_woql_ast('InsertDocument',JSON,WOQL,Path) :-
    _{document : Doc
     } :< JSON,
    json_value_to_woql_ast(Doc, WDoc, [document
                                       |Path]),
    WOQL = insert_document(WDoc).
```

`src/core/query/woql_compile.pl`:

```prolog
compile_wf(insert_document(Doc,X),(
               freeze(Guard,
                      insert_document(S0, DocE, URI)))) -->
    resolve(X,URI),
    resolve(Doc,DocE),
    view(update_guard, Guard),
    peek(S0).
```

**Comment by spl** [comment]
Fixed by #919.
